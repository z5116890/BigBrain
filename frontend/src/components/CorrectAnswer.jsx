import React from 'react';
import { Typography, Card, makeStyles } from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';
import CheckIcon from '@material-ui/icons/Check';
import PropTypes from 'prop-types';

// Shows user if their answer is correct or not through box border colour and tick or cross icon
function CorrectAnswer(props) {
  const {
    correct,
    answer,
  } = props;
  const useStyles = makeStyles({
    tick: {
      fill: 'green',
    },
    cross: {
      fill: 'red',
    },
    root: {
      border: correct ? 'green solid 1px !important' : 'red solid 1px !important',
      marginTop: '2px',
      width: '40%',
      height: '100%',
    },
    text: {
      textAlign: 'center',
      color: 'black',
    },
  });
  const classes = useStyles();

  return (
    <Card id="answer" className={classes.root}>
      <Typography data-test="answer" className={classes.text} variant="h5" component="h2">
        {answer}
        {correct
          ? (<CheckIcon id="tick" className={classes.tick} />)
          : (<CloseIcon id="cross" className={classes.cross} />)}
      </Typography>
    </Card>
  );
}
CorrectAnswer.propTypes = {
  correct: PropTypes.bool.isRequired,
  answer: PropTypes.string.isRequired,
};
export default CorrectAnswer;
