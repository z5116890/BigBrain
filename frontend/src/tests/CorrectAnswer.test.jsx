import React from 'react';
import renderer from 'react-test-renderer';
import { Typography } from '@material-ui/core';
import { createShallow } from '@material-ui/core/test-utils';
import CorrectAnswer from '../components/CorrectAnswer';

describe('CorrectAnswer', () => {
  let shallow;
  beforeAll(() => {
    shallow = createShallow({ dive: true });
  });
  it('should contain 1 Card component', () => {
    const wrapper = shallow(<CorrectAnswer correct answer="yes" />);
    expect(wrapper.find('#answer')).toHaveLength(1);
  });
  it('should contain 1 Typography component', () => {
    const wrapper = shallow(<CorrectAnswer correct answer="yes" />);
    expect(wrapper.find(Typography)).toHaveLength(1);
  });
  it('displays a custom correct answer given through props', () => {
    const wrapper = shallow(<CorrectAnswer correct answer="yes" />);
    const title = wrapper.find('[data-test="answer"]');
    expect(title.text()).toBe('yes');
  });
  it('displays a red cross if answer is incorrect', () => {
    const wrapper = shallow(<CorrectAnswer correct={false} answer="yes" />);
    expect(wrapper.find('#cross').exists()).toEqual(true);
  });
  it('displays a green tick if answer is correct', () => {
    const wrapper = shallow(<CorrectAnswer correct answer="yes" />);
    expect(wrapper.find('#tick').exists()).toEqual(true);
  });
  // Snapshots. Making sure component does not change unexpectedly by
  // comparing a reference snapshot with test snapshot.
  it('renders with props', () => {
    const ca = renderer.create(<CorrectAnswer correct answer="yes" />);
    expect(ca).toMatchSnapshot();
  });
});
