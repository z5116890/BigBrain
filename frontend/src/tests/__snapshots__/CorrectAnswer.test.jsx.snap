// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CorrectAnswer renders with props 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-23 MuiPaper-elevation1 MuiPaper-rounded"
  id="answer"
>
  <h2
    className="MuiTypography-root makeStyles-text-24 MuiTypography-h5"
    data-test="answer"
  >
    yes
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root makeStyles-tick-21"
      focusable="false"
      id="tick"
      viewBox="0 0 24 24"
    >
      <path
        d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
      />
    </svg>
  </h2>
</div>
`;
